dbms/src/Columns/ColumnVector.cpp

```diff
@@ -314,6 +314,35 @@ ColumnPtr ColumnVector<T>::replicate(const IColumn::Offsets & offsets) const
     return res;
 }
 
+template <typename T>
+ColumnPtr ColumnVector<T>::replicate(size_t start_row, size_t end_row, size_t prev_offset, const IColumn::Offsets & offsets) const
```

Comment:
Why we need pass `prev_offset` as an argument? In my understanding `prev_offset = start_row == 0 ? 0 `: offsets[start_row -1]?

Reply:
done.